### DISCUSSION TITLE
reading from a text file to produce frames of a crude animation?

### DISCUSSION THREAD

**Discussion by Twixtfanatic** [ImageMagick/ImageMagick#2364] [Help]
I want to use two svg files, a text file, and an MSL script file to generate the **frames** of a crude animation which is like a South Park style talking head. One svg file represents the top half of a 2-D head with an otherwise transparent background, which will move around inside the larger rectangular svg image. I want the MSL script to accept the parameters height and width of the larger rectangular image in pixels. I want the script to read numerical values from a text file, a group of N values at a time. These N values describe the location of the upper head inside the box for a specific frame of animation: The location of the center of rotation relative to the upper head, the amount of rotation, and the location of this center of rotation relative to the larger box. That's one angle value, 2 relative x values which range from 0 to 1, and 2 relative y values which range from 0 to 1. For example the upper head might pivot  about its lower left, or it might pivot around its lower right, or it might move straight up and down over the lower half of the head. The relative scale of the upper head within the larger image will remain constant. The composite (and maybe also the convert) command with the -distort option could be used to combine these two svg images to make a png image in the desired resolution. For example, if the animation lasts 10 seconds at a frame rate of 60 fps and each frame requires 5 numerical values to define exactly how the upper head is composited, the text file would contain 3000 numerical values, separated by commas, which the script would read to produce 600 frames of animation.

I do NOT want to create an animation with Image Magick, just frames of the form videoname_framenumber.png which will become part of a much larger video. Does this all sound possible? I'm not asking you to write my code, just please tell me if I'm on the right track. Can MSL do this? Thanks very much.

**Comment by snibgo** [comment]
I have never used MSL for anything.

MSL is documented at https://www.imagemagick.org/script/conjure.php . It says "Any element with a strike-thru is not supported yet." About half the elements have a strike-thru.

In addition, the examples at the top of that page don't work for me.

So, if MSL works for you and does what you need, that's fine. But I wouldn't recommend it.

Instead, I suggest you generate either IM v7 scripts (that are called with the IM operation "-script") or IM CLI commands in shell scripts that are called with the usual shell methods.

EDIT: Another aspect to consider: your sources are SVG files. You want raster files as output. You want to translate and rotate the inputs. You might translate and rotate within the SVG files so the rasterizer directly creates the outputs you want. Alternatively, you might rasterize the original SVG files, then apply translation and rotation to those. This second method might give lower quality, but you could supersample the rasterizing (eg rasterize at 200% of your final required dimensions) and shrink after the transformation.
